packages/web/src/components/comments/CommentBlock.tsx

```diff
@@ -0,0 +1,192 @@
+import { useState } from 'react'
+
+import { ID } from '@audius/common/models'
+import {
+  Avatar,
+  Flex,
+  IconButton,
+  IconHeart,
+  IconMerch,
+  IconPencil,
+  IconTrash,
+  Text,
+  TextLink
+} from '@audius/harmony'
+
+import { decodeHashId } from 'utils/hashIds'
+
+import { CommentForm } from './CommentForm'
+import { useCurrentCommentSection } from './CommentSectionContext'
+import type { Comment } from './types'
+
+// TODO: do we need hours?
+const formatTimestampS = (timestamp_s: number) => {
+  const hours = Math.floor(timestamp_s / (60 * 60))
+  const minutes = Math.floor(timestamp_s / 60)
+  const seconds = `${timestamp_s % 60}`.padStart(2, '0')
+  if (hours > 0) {
+    return `${hours}:${minutes}:${seconds}`
+  } else {
+    // TODO: check designs- I think we want a m:ss format regardless
+    return `${minutes}:${seconds}`
+  }
+}
+
+export type CommentBlockProps = {
+  comment: Comment
+  parentCommentId?: ID
+  parentCommentIndex?: number
+}
+
+export const CommentBlock = ({
+  comment,
+  parentCommentId, // Parent comment ID can be passed in order to reply to a reply, otherwise it's assumed you're replying to this comment
+  parentCommentIndex
+}: CommentBlockProps) => {
```

Comment:
Is there a specific reason for that? I'm not against, just curious cause I generally prefer this method

Reply:
I personally prefer destructing in the args too but there are some good reasons to destructure in the component!
1. Sometimes you need access to certain props *and* the entire `props` object in a component. For example, to pass `...props` to a child
2. Components that only optionally accept props can't destructure in the arguments
3. It's easier to refactor to a forwardRef component, and a little bit easier to parse the first few lines of the component

Reply:
Fair, but I'm not quite sure what you mean by `2`?
Do you mean using like a union type or something?
like 
```ts
type Props = {} | { thing: number }
```

Reply:
Actually now that I think about it #2 doesn't really apply to components, only functions. I think we just wanted to go for consistency because of the possibility of #1

Let's talk about it during the frontend sync!